main.js

```diff
@@ -0,0 +1,329 @@
+  let fiveDayForecastDays = [];
+  let allObjForecastEntriesArr = [];
+
+  data.list.forEach(function(item){
+    let dayOfWeek = getNameOfDay(item);
+    let dayObj = {
+      city: data.city.name,
+      dayOfWeek: dayOfWeek,
+      temp: Math.floor(item.main.temp),
+      clouds: item.weather[0].main
+    }
+    allObjForecastEntriesArr.push(dayObj);
+  });
+
+  fiveDayForecastDays = function(objArray){
+    return objArray.reduce(function(acc, dayOfWeek){
+      let exists = acc.find(function (item){
+        return dayOfWeek.dayOfWeek === item.dayOfWeek;
+      });
+      if(!exists){
+        acc.push(dayOfWeek);
+      }
+      return acc;
+    }, []);
+  }
+  return fiveDayForecastDays(allObjForecastEntriesArr);
+}
+
+/**
+ * 
+ * @param {JSON} item - item object from JSON object promise
+ * @returns {String} - string name of day of week from JSON object
+ */
+function getNameOfDay (item){
+  let dayNumIndex = new Date(item.dt*1000).getDay();
+  return dayNameNumberIndicesObj[dayNumIndex];
+}
+
+/**
+ * 
+ * @param {Array} tempsArray 
+ * @returns {Number} - average of all number items in @param tempsArray
+ */
+
+function avgTempForDay(tempsArray){
+  let avgTemp = tempsArray.reduce(function(sum, temp){
+    return sum + temp;
+  }, 0);
+  return Math.floor(avgTemp/tempsArray.length);
+}
+
+/**
+ * 
+ * @param {Array} cloudsArray 
+ * @returns {Number} - number for highest count item in @param {Array} cloudsArray 
+ */
+function avgCloudCoverForDay(cloudsArray){
+  let forecastObjCount = cloudsArray.reduce(function(acc, forecast){
+    if(acc.hasOwnProperty(forecast)){
+      acc[forecast] += 1;
+    } else {
+      acc[forecast] = 1;
+    }
+    return acc;
+  }, {});
+
+  let highestCount = Math.max(...Object.values(forecastObjCount));
+
+  return Object.keys(forecastObjCount).find(function(key){
+    return forecastObjCount[key] === highestCount;
+  });
+}
+
+/**
+ * 
+ * @param {String} forecastSky 
+ * @returns {String} - string representation of DOM element for weather forecast icon image representation - Big version
+ */
+let selectBigWeatherIcon = function(forecastSky){
+  switch(true) {
+    case(forecastSky === 'Clear'):
+      return forecastSky = sunElement;
+    case(forecastSky === 'Clouds'):
+      return forecastSky = cloudElement;
+    case(forecastSky === 'Rain'):
+      return forecastSky = rainElement;
+    case(forecastSky === 'Snow'):
+      return forecastSky = snowElement;
+    case(forecastSky === 'Mist'):
+      return forecastSky = mistElement;
+    case(forecastSky === 'Smoke'):
+      return forecastSky = smokeElement;
+    default:
+      return forecastSky = nullWeatherElement; 
+  }
+}
+/**
+ * 
+ * @param {String} fiveDayForecast 
+ * @returns {String} - string representation of DOM element for weather forecast icon image representation - Small version
+ */
+let selecSmallWeatherIcon = function(fiveDayForecast) {
+  switch(true) {
+    case(fiveDayForecast === 'Clear'):

```

Comment:
this function seems repetitive too - could be combined with the above and just do a condition before the switch to determine what type of forecast it is.